File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/RadioButton/Functionality/CheckedPage.vb
Namespace CompanionKit.Controls.RadioButton.Functionality

	Public Class CheckedPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

			'Update label text with initial state of demonstrated RadioButton
			Me.UpdateLabelText()


		End Sub

        Private Sub mobjButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjButton.Click
            'Updates label text, that indicates state of demonstrated CheckBox.
            Me.UpdateLabelText()
        End Sub

		''' <summary>
		''' Updates label text, that indicates state of demonstrated RadioButton.
		''' </summary>
		''' <remarks></remarks>
		Private Sub UpdateLabelText()
            Me.mobjInfoLabel.Text = String.Format("RadioButton1 is {0}!", IIf(Me.mobjRadioButton1.Checked, "checked", "unchecked"))
		End Sub

	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/RadioButton/Functionality/CheckedPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.RadioButton.Functionality
{
    public partial class CheckedPage : UserControl
    {
        public CheckedPage()
        {
            InitializeComponent();
            //Update label text with initial state of demonstrated RadioButton
            UpdateLabelText();
        }

        /// <summary>
        /// Updates label text, that indicates state of demonstrated RadioButton.
        /// </summary>
        private void UpdateLabelText()
        {
            this.mobjInfoLabel.Text = String.Format("RadioButton1 is {0}!", (this.mobjRadioButton1.Checked ? "checked" : "unchecked"));
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Updates label text, that indicates state of demonstrated CheckBox.
            UpdateLabelText();
        }
    }
}